hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2316,16 +2318,28 @@ public List<OzoneFileStatusLight> listStatusLight(String volumeName,
       String bucketName, String keyName, boolean recursive, String startKey,
       long numEntries, boolean allowPartialPrefixes) throws IOException {
     OmKeyArgs keyArgs = prepareOmKeyArgs(volumeName, bucketName, keyName);
-    if (omVersion.compareTo(OzoneManagerVersion.LIGHTWEIGHT_LIST_STATUS) >= 0) {
-      return ozoneManagerClient.listStatusLight(keyArgs, recursive, startKey,
-          numEntries, allowPartialPrefixes);
-    } else {
-      return ozoneManagerClient.listStatus(keyArgs, recursive, startKey,
-              numEntries, allowPartialPrefixes)
-          .stream()
-          .map(OzoneFileStatusLight::fromOzoneFileStatus)
-          .collect(Collectors.toList());
+    if (omVersion.compareTo(OzoneManagerVersion.MAYBE_LIGHTWEIGHT_LIST_STATUS) >= 0) {
+      try {
+        return ozoneManagerClient.listStatusLight(keyArgs, recursive, startKey,
```

Comment:
Maybe we can add caching of whether OM supports `listStatusLight`, the current implementation would at worst probably make every one `listStatusLight` at least twice RPC with the OM.

Reply:
Good idea.  Please note that both version 4 (ad176eb0c10ced990d2063cb1b6d4004c77dbe55) and `listStatusLight` (6faf0023b057dc2cc51bdbf288a43e756fb42b4b) were both introduced in 1.4.0.  Fallback is necessary only for those using Ozone based on any of the 118 commits in between these two.